R/distribution_formulas.R

```diff
@@ -0,0 +1,192 @@
+    stop("use family = gaussian() instead")
+  }
+  if (!(family[["family"]] %in% families)) {
+    stop("FIMS currently does not offer this distribution.")
+  }
+  if ((data_type == "agecomp" | data_type == "lengthcomp") &
+      (family[["family"]] == "lognormal" | family[["family"]] == "gaussian")) {
+    stop("Did you mean family = multinomial()?")
+  }
+  if ((data_type == "index" | data_type == "cpue") &
+      family[["family"]] == "multinomial") {
+    stop("Multinomial is not available for index or CPUE data")
+  }
+
+  if (data_type == "index" | data_type == "cpue") {
+    obs_id_name <- "observed_index_data_id"
+  }
+  if (data_type == "agecomp" | data_type == "lengthcomp") {
+    obs_id_name <- "observed_agecomp_data_id"
+  }
+  if (family[["family"]] == "lognormal") {
+    new_module <- new(TMBDlnormDistribution)
+    new_module$log_logsd <- new(
+      ParameterVector,
+      log(sd$value),
+      length(sd$value)
+    )
+    new_module$log_logsd$set_all_estimable(sd$estimated)
+    if (family$link == "log") {
+      expected <- "log_expected_index"
+    }
+    if (family$link == "identity") {
+      expected <- "expected_index"
+    }
+  }
+
+  if (family[["family"]] == "gaussian") {
+    new_module <- new(TMBDnormDistribution)
+    new_module$log_sd <- new(ParameterVector, log(sd$value), length(sd$value))
+    new_module$log_sd$set_all_estimable(sd$estimated)
+    if (family$link == "log") {
+      expected <- "log_expected_index"
+    }
+    if (family$link == "identity") {
+      expected <- "expected_index"
+    }
+  }
+
+  if (family[["family"]] == "multinomial") {
+    new_module <- new(TMBDmultinomDistribution)
+    expected <- "proportion_catch_numbers_at_age"
+  }
+  if (data_type == "index" | data_type == "cpue") {
+    new_module$set_observed_data(module$GetObservedIndexDataID())
+  }
+  if (data_type == "agecomp") {
+    new_module$set_observed_data(module$GetObservedAgeCompDataID())
+  }
+
+  new_module$set_distribution_links("data", module$field(expected)$get_id())
+  return(new_module)
+}
+
+#' Sets up a new distribution for a process
+#' @param par A string specifying the parameter name the distribution applies
+#'   to.
+#' @inheritParams new_data_distribution
+#' @param is_random_effect A boolean indicating whether or not the process is
+#'   estimated as a random effect.
+#' @seealso
+#' * [new_data_distribution()]
+#' @export
+new_process_distribution <- function(par,
+                                     module,
+                                     family,
+                                     sd = list(value = 1, estimated = FALSE),
+                                     is_random_effect = FALSE) {
+  families <- c("lognormal", "gaussian")
+  if (family[["family"]] == "normal") {
+    stop("use family = gaussian() instead")
+  }
+  if (!(family[["family"]] %in% families)) {
+    stop("FIMS currently does not offer this distribution for processes.")
+  }
+  if (family[["family"]] == "lognormal") {
```

Comment:
Before this line it would be good to have a comment delineating that you are actually setting up the distribution here based on the type passed. That way it is clear where code should be added if a new distribution is allowed. Leave some breadcrumbs for the future reader and developer of the code. This same comment pertains to line 44 as well.

Reply:
added comments